examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,430 @@
+# because of their different unit of measure.
+# For instance, the AGE coefficient is expressed in $/hours/living years
+# while the EDUCATION is expressed in $/hours/years of education.
+# This is evident if we compare feature standard deviations.
+
+X_train_preprocessed = pd.DataFrame(
+    model.named_steps['columntransformer'].transform(X_train),
+    columns=feature_names
+)
+X_train_preprocessed.std(axis=0).plot(kind='barh', figsize=(9, 7))
+plt.title('Features std. dev.')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# We should then multiply the coefficients by the standard deviation and we
+# will be able to better compare them: in that way, we emphasize that the
+# greater the variance of a feature, the larger the impact of the corresponding
+# coefficent on the output, all else being equal.
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_ *
+    X_train_preprocessed.std(axis=0),
+    columns=['Coefficients'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# The plot above tells us that an increase of the AGE will induce a decrease
+# of the WAGE when all other features remain constant, for instance at a
+# constant EXPERIENCE. An increase of the EXPERIENCE will induce an increase
+# of the WAGE when all other features remain constant.
+#
+# Checking the coefficient variability
+# ....................................
+#
+# We can check the coefficient variability through cross-validation.
+# If coefficients vary in a significant way changing the input dataset
+# the robustness of the model is not guaranteed.
+
+from sklearn.model_selection import cross_validate
+from sklearn.model_selection import RepeatedKFold
+
+cv_model = cross_validate(
+    model, X, y, cv=RepeatedKFold(n_splits=5, n_repeats=5),
+    return_estimator=True, n_jobs=-1
+)
+coefs = pd.DataFrame(
+    [est.named_steps['transformedtargetregressor'].regressor_.coef_ *
+     X_train_preprocessed.std(axis=0)
+     for est in cv_model['estimator']],
+    columns=feature_names
+)
+plt.figure(figsize=(9, 7))
+sns.swarmplot(data=coefs, orient='h', color='k', alpha=0.5)
+sns.boxplot(data=coefs, orient='h', color='cyan', saturation=0.5)
+plt.axvline(x=0, color='.5')
+plt.title('Coefficient variability')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# The AGE and EXPERIENCE coefficients are affected by strong variability which
+# might be due to the collinearity between the 2 features.
+# To verify this interpretation we plot the variability of the AGE and
+# EXPERIENCE coefficient:
+
+plt.ylabel('Age coefficient')
+plt.xlabel('Experience coefficient')
+plt.grid(True)
+plt.scatter(coefs["AGE"], coefs["EXPERIENCE"])
```

Comment:
Maybe a title on this plot, to help people scanning through the example fast. For instance:
```suggestion
plt.scatter(coefs["AGE"], coefs["EXPERIENCE"])
plt.title('Variations of coefficients for AGE and EXPERIENCE across folds')
```